File: searching algoritms/bubble_sort.rb
def bubble_sort(arr)
  n = arr.length

  while n > 0 do
    n -= 1
    make_switch = false
    
    n.times do | i |
      if arr[i] > arr[i + 1]
        temp = arr[i]
        arr[i] = arr[i + 1]
        arr[i + 1] = temp
        make_switch = true
      end
    end
    print arr
    break unless make_switch
  end

  arr
end


p bubble_sort([1, 4, 3, 2, 5])

File: searching algoritms/bubble-sort.js
let bubbleSort = (arr) => {
  for(let i = arr.length; i > 0; i --) {
    console.log(arr)
    let makeSwitch = false;
    for(let j = 0; j < i; j ++) {
      if(arr[j] > arr[j + 1]) {
        [arr[j], arr[j + 1]] = [arr[j + 1], arr[j]]
        makeSwitch = true
      }
    }

    if(!makeSwitch) break;
    
  }
  return arr
}

// [1, 3, 2, 4, 5]

console.log(bubbleSort([1,4,3,2,5,8,9,7]))